packages/core/src/components/editable-text/editableText.tsx

```diff
@@ -196,7 +196,7 @@ export class EditableText extends AbstractPureComponent<IEditableTextProps, IEdi
 
     public componentDidUpdate(_: IEditableTextProps, prevState: IEditableTextState) {
         if (this.state.isEditing && !prevState.isEditing) {
-            safeInvoke(this.props.onEdit);
+            safeInvoke(this.props.onEdit, prevState.value);
```

Comment:
I _feel_ like passing `prevState` is wrong in `componentDidUpdate`, as it's no longer an accurate picture of the world. what happens if we use `this.state.value` instead?

Reply:
*To be honest*, I really don't see any difference between `prevState` and `this.state` **in this case**. As the condition `this.state.isEditing && !prevState.isEditing` will be true only and only if the user triggered `onFocus` event; other than that, it will be always false (*As `toggleEditing()` is being called in 3 possible locations, all of which are to change `isEditing` from `true` to `false`. So the only function that will change `isEditing` from `false` to `true` is `handleFocus()`*).

The point here is that when `safeInvoke(this.props.onEdit, prevState.value)` is invoked the value of `prevState.value` and `this.state.value` will be the same, so there's nothing wrong neither with `prevState.value` nor `this.state.value`.

And the reason behind choosing `prevState.value` over `this.state.value` is that we want the value of `Editable-text` before it gets re-rendered which means before it gets modified; but as I explained before .. there's nothing wrong with the other alternative, it's just about the logic of thinking as using `prevState` would sound more logical than using `state`.

Notice that I may have missed something; if it's the case, please correct me.

Have a nice day 😁

Reply:
I see now what you're trying to prove here .. And I'm 100% with you in that.

I'll be committing the needed change right away.

And thank you for your time, have a nice day ..